assembly/src/compiled_library.rs

```diff
@@ -0,0 +1,184 @@
+use alloc::{collections::BTreeMap, vec::Vec};
+use vm_core::{
+    crypto::hash::RpoDigest,
+    mast::{MastForest, MerkleTreeNode},
+};
+
+use crate::{
+    ast::{FullyQualifiedProcedureName, ProcedureIndex, ProcedureName},
+    CompiledLibraryError, LibraryNamespace, LibraryPath, Version,
+};
+
+/// A procedure's name, along with its module path.
+///
+/// The only difference between this type and [`FullyQualifiedProcedureName`] is that
+/// [`CompiledFullyQualifiedProcedureName`] doesn't have a [`crate::SourceSpan`].
+pub struct CompiledFullyQualifiedProcedureName {
```

Comment:
You can definitely just use `FullyQualifiedProcedureName` instead of this, just construct it with a default `SourceSpan` if you don't have one. In most cases though, we usually have some kind of context for which a span can be obtained/derived.

Reply:
Actually, you probably can just use `LibraryPath` - a `LibraryPath` can always be converted to a `FullyQualifiedProcedureName` (by splitting the last component off). The only reason to use the `FullyQualifiedProcedureName` is when you want convenient access to both the module and procedure name without having to deal with extracting them from a raw `LibraryPath`

Reply:
As discussed in other PR, we now use `FullyQualifiedProcedureName`